libr/core/agraph.c

```diff
@@ -4046,6 +4038,116 @@ static bool toggle_bb(RCore *core, ut64 addr) {
 	return false;
 }
 
+typedef struct r_canvas_location {
+	int x;
+	int y;
+} RPosition;
+
+static const char *strstr_xy(const char *p, const char *s, int *x, int *y) {
+	const char *d = strstr (p, s);
+	if (!d) {
+		return NULL;
+	}
+	const char *q;
+	int nl = 0;
+	int nc = *x;
+	for (q = p; q < d; q++) {
+		if (*q == '\n') {
+			nl++;
+			*x = 0;
+			nc = 0;
+		} else {
+			nc++;
+		}
+	}
+	*x += nc;
+	*y += nl;
+	return d;
+}
+
+static char *get_graph_string(RCore *core) {
+	int c = r_config_get_i (core->config, "scr.color");
+	int u = r_config_get_i (core->config, "scr.utf8");
+	r_config_set_i (core->config, "scr.color", 0);
+	r_config_set_i (core->config, "scr.utf8", 0);
+	r_core_visual_graph (core, NULL, NULL, false);
+	char *s = strdup (r_cons_get_buffer ());
+	r_cons_reset ();
+	r_config_set_i (core->config, "scr.color", c);
+	r_config_set_i (core->config, "scr.utf8", u);
+	return s;
+}
+
+static void nextword(RCore *core, RConsCanvas *can, const char *word) {
+	r_return_if_fail (core && core->graph && can && word);
+	if (!*word) {
+		return;
+	}
+	RCoreGraphHits *gh = &core->graph->ghits;
+	if (gh->word_list && gh->old_word && !strcmp (word, gh->old_word)) {
+		RPosition *pos = r_list_get_n (gh->word_list, gh->word_nth);
+		if (pos) {
+			gh->word_nth++;
+		} else {
+			gh->word_nth = 0;
+			pos = r_list_get_n (gh->word_list, gh->word_nth);
+		}
+		if (pos) {
+			can->sx = pos->x;
+			can->sy = pos->y;
+		}
+		return;
+	} else {
+		r_list_free (gh->word_list);
+		gh->word_list = r_list_newf (free);
+	}
+	char *s = get_graph_string (core);
+	int x, y;
+	const char *p = s;
+	r_cons_clear00 ();
+	r_cons_flush ();
+	int ox = 0;
+	int oy = 0;
+	size_t count = 0;
+	const size_t MAX_COUNT = 4096;
+	x = y = 0;
+	bool first_x = true;
+	while (true) {
```

Comment:
convert this into something like `while (a && count < COUNT_MAX)` and adjust the body.